twigs/ecr.py

```diff
@@ -0,0 +1,172 @@
+import sys
+import os
+import subprocess
+import logging
+import json
+
+from . import docker
+
+
+g_encoding = None
+
+def set_encoding(encoding):
+    global g_encoding
+    g_encoding = encoding
+
+def get_encoding():
+    global g_encoding
+    return g_encoding
+
+def get_repo_uri(repository, repositoryType):
+
+    if repositoryType == 'private':
+        repo_cmd = "ecr describe-repositories"
+    else:
+        repo_cmd = "ecr-public describe-repositories"
+
+    t_json = run_aws_cmd(repo_cmd)
+
+    if len(t_json) > 0:
+        for i in t_json['repositories']:
+            if i['repositoryName'] == repository:
+                return i['repositoryUri']
+
+def run_aws_cmd(cmd):
+    cmd = 'aws --output json ' + cmd
+    try:
+        cmd_output = subprocess.check_output([cmd], shell=True, stdin=None, stderr=None)
+        cmd_output = cmd_output.decode(get_encoding())
+        ret_json = json.loads(cmd_output)
+
+    #except ClientError as e:
```

Comment:
Can we remove the commented block of code?

Reply:
removed. kept there to check whether I can catch any specific exception, then realized it is a call to subprocess.